### ISSUE TITLE
twisted.web.static.FileTransfer pointlessly uses threadable.synchronized

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2985]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2985|
|Type|defect|
|Created|2008-01-08 22:43:47Z|
|Branch|https://github.com/twisted/twisted/tree/unsynchronized-2985|

```FileTransfer``` should not bother to pretend to provide a threadsafe API.  Synchronizing ```resumeProducing``` doesn't actually make it threadsafe to call ```self.request.write``` from another thread.

<details><summary>Searchable metadata</summary>

```
trac-id__2985 2985
type__defect defect
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__branches_unsynchronized_2985 branches/unsynchronized-2985
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1199832227000000 1199832227000000
changetime__1213744769000000 1213744769000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [23910]) Branching to 'unsynchronized-2985'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Removed, and did some other minor cleanups.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

Looks cool. There is a bunch of other cleanups which would be nice to make:
 * remove string import
 * remove future import
 * remove trailing whitespaces

Nothing mandatory though, so merge when you want.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [23946]) Merge unsynchronized-2985

Author: exarkun
Reviewer: therve
Fixes: [#2985](https://github.com/twisted/twisted/issues/2985)

Remove the code which synchronizes methods of FileTransfer.  The methods
aren't threadsafe even if they're guaranteed to be serialized.

Also make some minor cleanups to nearby code, removing trailing whitespace
and no-longer needed imports.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
